recipes/protobuf/all/test_package/CMakeLists.txt

```diff
@@ -1,16 +1,16 @@
 cmake_minimum_required(VERSION 3.1)
 project(test_package)
 
-include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
-conan_basic_setup(TARGETS)
-
-find_package(protobuf CONFIG REQUIRED)
-
 find_program(PROTOC_PROGRAM protoc)
 if (NOT PROTOC_PROGRAM)
     message(WARNING "Protoc was not found")
 endif()
 
+include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
+conan_basic_setup(TARGETS)
+
+find_package(protobuf CONFIG REQUIRED)
+
```

Comment:
Maybe I missed it, but why did you move this section?
How protoc will be found, if your cmake is empty in the beginning and PATH is not set?

Reply:
Exactly, there is no reason for moving it.

Reply:
This one is needed though, that I'm sure of (and have double checked).
`find_program(PROTOC_PROGRAM protoc)` needs to go before `conan_basic_setup`
otherwise cmake will find the wrong protoc (host instead of build).
The protoc CMake should find comes from `self.env_info.PATH.append(bindir)`.

Reply:
Dug a bit deeper and I think a nicer patch is possible by playing around with cmake find paths here: https://github.com/conan-io/conan-center-index/blob/master/recipes/protobuf/all/conanfile.py#L89

Reply:
What do you think about: https://github.com/conan-io/conan-center-index/pull/4556/commits/3b26749c4689b81fc9c5e45f0659d6b64ec9ce1f?